doc/port_auto_neg/port-auto-negotiation-design.md

```diff
@@ -0,0 +1,455 @@
+# SONiC Port Auto Negotiation Design #
+
+## Table of Content 
+
+### Revision ###
+
+ | Rev |     Date    |       Author       | Change Description                |
+ |:---:|:-----------:|:------------------:|-----------------------------------|
+ | 0.1 |             |      Junchao Chen  | Initial version                   |
+
+### Scope
+This document is the design document for port auto negotiation feature on SONiC. This includes the requirements, CLI change, DB schema change, swss change.
+
+### Definitions/Abbreviations 
+N/A
+ 
+### Overview
+
+The IEEE 802.3 standard defines a set of Ethernet protocols that are comprised of speed rate and interface type. It allows for configuring multiple values at the same time for port provisioning and advertising to the remote side. However, on SONiC, user can configure the speed of port, and user can configure auto negotiation mode via config DB. Port attributes such as interface type, advertised speeds, advertised interface types are not supported.
+
+The feature in this document is to address the above issues.
+
+### Requirements
+
+The main goal of this document is to discuss the design of following requirement:
+
+- Allow user to configure auto negotiation via CLI
+- Allow user to configure advertised speeds
+- Allow user to configure interface type
+- Allow user to configure advertised interface types
```

Comment:
can you add link training into the requirement. enable/disable link training.

Reply:
Need collect information, and this will be added to phase 2.